test/list.vector.spec.js

```diff
@@ -0,0 +1,109 @@
+/* eslint-env mocha */
+
+const Block = require('@ipld/block')
+const assert = require('assert')
+const Vector = require('../src/lists/vector')
+const { Lookup, get } = require('../')
+const getPath = get
+
+function storage () {
+  let kv = new Map()
+  return {
+    async put (block) {
+      return kv.set((await block.cid()).toString(), block)
+    },
+    get (cid) {
+      return kv.get(cid.toString())
+    },
+    lookup // so we can pass to get()
+  }
+}
+
+async function asyncList (iter) {
+  let parts = []
+  for await (let part of iter) {
+    parts.push(part)
+  }
+  return parts
+}
+
+const lookup = new Lookup()
+lookup.register(Vector._type, Vector)
+
+describe('Vector', () => {
+  let fixture = Array.from({ length: 301 }).map((v, i) => { return { leafValue: `v${i}` } })
+  let blocks = fixture.map(b => Block.encoder(b, 'dag-cbor'))
+  let cids
+
+  before(async () => {
+    cids = await Promise.all(blocks.map(b => b.cid()))
+  })
+
+  it('create with CID values', async () => {
+    let iter = Vector.create(cids, 3)
+    let trace = await asyncList(iter)
```

Comment:
What is the reason to use `asyncList`? Couldn't you just change the for loop below to

```JS
for await (const block of iter) { …  }
```